lib/lvol/lvol.c

```diff
@@ -1378,6 +1433,124 @@ spdk_lvol_create_snapshot(struct spdk_lvol *origlvol, const char *snapshot_name,
 				lvol_create_cb, req);
 }
 
+struct xattrs_ctx {
+	struct spdk_lvol *newlvol;
+	const char *const *xattrs_external;
+};
+
+static void
+lvol_get_xattr_value_ext(void *xattr_ctx, const char *name,
+			 const void **value, size_t *value_len)
+{
+	struct xattrs_ctx *ctx = xattr_ctx;
+	size_t i;
+
+	lvol_get_xattr_value(ctx->newlvol, name, value, value_len);
+
+	if (*value == NULL || *value_len == 0) {
+		/* Search for external xattr*/
+		for (i = 0; ctx->xattrs_external[i]; i++) {
+			/* Skip xattrs value, we search xattr names, which are in even positions */
+			if (i % 2 != 0) {
+				continue;
+			}
+
+			if (strcmp(name, ctx->xattrs_external[i]) == 0) {
+				*value = ctx->xattrs_external[i + 1];
+				*value_len = strlen(ctx->xattrs_external[i + 1]) + 1;
+				return;
+			}
+		}
+
+		*value = NULL;
+		*value_len = 0;
+	}
+}
+
+void
+spdk_lvol_create_snapshot_xattrs(struct spdk_lvol *origlvol, const char *snapshot_name,
```

Comment:
```suggestion
spdk_lvol_create_snapshot_with_xattrs(struct spdk_lvol *origlvol, const char *snapshot_name,
```
spdk_lvol_create_snapshot_xattrs sounds only creating xattrs